File: BEST-App-main/lib/model/wishlist.dart
class Wishlist {
  final int id;
  final int productId;
  final int userId;
  final String createdAt;
  final String updatedAt;

  Wishlist(this.id, this.productId, this.userId, this.createdAt,
      this.updatedAt);

  static Map<String, dynamic> toJson(Wishlist wishlist) {
    return {
      'id': wishlist.id,
      'product_id': wishlist.productId,
      'user_id': wishlist.userId,
      'createdAt': wishlist.createdAt,
      'updatedAt': wishlist.updatedAt
    };
  }
}

File: spring-best/src/main/java/com/best/springbest/model/Wishlist.java
package com.best.springbest.model;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "wishlists")
public class Wishlist {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "product_id",nullable = false)
    private Product product;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "user_id",nullable = false)
    private User user;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date created_at;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date updated_at;

}
